app/assets/stylesheets/local/chf_image_viewer.scss

```diff
@@ -0,0 +1,315 @@
+// Override some bootstrap modal stuff to give us a full-viewport modal
+#chf-image-viewer-modal {
+  .modal-dialog {
+    position: fixed;
+    height: 100%;
+    width: 100%;
+    margin: 0;
+    padding: 0;
+    border: 0;
+  }
+  .modal-content {
+    border: 0 none;
+    border-radius: 0;
+  }
+
+  // custom slide-up animation on open, no fade.
+  .modal-dialog {
+    top: 100vh;
+    bottom: 200vh;
+    transition-property: top, bottom;
+    transition-duration: 0.5s;
+    transition-timing-function: ease-in-out;
+  }
+  &.in .modal-dialog {
+    top: 0;
+    bottom: 100vh;
+  }
+
+  //@include translate(0, -100%);
+  //@include transition-transform(0.3s ease-out);
+  //&.in .modal-dialog { @include translate(0, 0) }
+}
+
+.chf-image-viewer {
+  //cover whole screen. not totally sure why we need to repeat this
+  //when wrapping modal-dialog is already, but it works.
+  position: fixed;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  max-width: 100vw;
+
+  background-color: $body-bg;
+
+  display: flex;
+  flex-direction: column;
+
+  .viewer-header {
+    flex-shrink: 0;
+    background-color: $brand-inverse;
+    // this should be a variable, or really we should have a mixin for the whole header nav style
+    color: $brand-primary;
+    a {
+      color: $brand-primary;
+      &:hover {
+        color: white;
+      }
+    }
+
+    display: flex;
+    align-items: center;
+    padding: 4px 15px;
+
+    .chf-logo {
+      width: 50px;
+    }
+
+    h1 {
+      flex-grow: 1;
+
+      padding: 0;
+      margin: 0 15px;
+
+      font-size: $font-size-h3;
+
+      overflow: hidden;
+
+      a {
+        display: block;
+        white-space: nowrap;
+        overflow: hidden;
+        text-overflow: ellipsis;
```

Comment:
i also used it in child-work names on the show page thumbs, to restrict em to one line.